usim/_primitives/context.py

```diff
@@ -115,36 +138,77 @@ def do(
                 await containing_scope
                 print('... scope has finished')
 
-            with Scope() as scope:
+            async with Scope() as scope:
                 scope.do(graceful(scope))
 
-        All `volatile` activities are aborted at the end of the scope,
-        after all non-`volatile` activities have finished.
-        Aborting ``volatile` activities is not graceful:
+        All ``volatile`` activities are aborted at the end of the scope,
+        after all non-``volatile`` activities have finished.
+        Aborting ``volatile`` activities is not graceful:
         :py:class:`GeneratorExit` is raised in the activity,
-        which must exit without `await`\ ing or `yield`\ ing anything.
+        which must exit without ``await``\ ing or ``yield``\ ing anything.
         """
-        child_task = Task(payload)
+        assert after is None or at is None,\
+            "schedule date must be either absolute or relative"
+        assert after is None or after > 0,\
+            "schedule date must not be in the past"
+        assert at is None or at > __LOOP_STATE__.LOOP.time,\
```

Comment:
Same as above, we should consider allowing to specify the current time.

```suggestion
        assert at is None or at >= __LOOP_STATE__.LOOP.time,\
```

Reply:
This is an artefact of how the event loop works - the current time is just "no time specified" to avoid checks. The event ``time + 0`` is not valid either.

We could change that though - the overhead is entirely negligible. Care to open a ticket?